src/indexer.c

```diff
@@ -395,12 +398,16 @@ int IndexDocument(RSAddDocumentCtx *aCtx) {
  * This helps keep Redis responsive during long indexing operations.
  * @param ctx The Redis context
  */
-void IndexerYield(RedisModuleCtx *ctx) {
+void IndexerYieldWhileLoading(RedisModuleCtx *ctx) {
   static size_t opCounter = 0;
-  
-  // Yield to Redis every RSGlobalConfig.indexerYieldEveryOps operations
-  if (++opCounter >= RSGlobalConfig.indexerYieldEveryOps) {
+
+  // If server is loading, Yield to Redis every RSGlobalConfig.indexerYieldEveryOps operations
+  if (RedisModule_Yield
+   && RedisModule_GetContextFlags(ctx) & REDISMODULE_CTX_FLAGS_LOADING
```

Comment:
Can we avoid testing this here? This function cannot be inlined, so it will significantly affect performance. We should test it once higher up and propagate the decision here